### PR TITLE
Send output files as a single zip

### PR DESCRIPTION
**Pull request by sebastienros**
Instead of single individual files using gzip.

### CODE REVIEW

**Comment by trylek** [comment]
Thanks @sebastienros for making the change; it looks great to me, I have approved it but apparently I don't have write access to the crank repo so that the automated merge didn't happen; I'm afraid you might need to identify another reviewer with write access.

File: src/Microsoft.Crank.Agent/Controllers/JobsController.cs
Comment (mkArtakMSFT) [inline src/Microsoft.Crank.Agent/Controllers/JobsController.cs:357]: I agree that it's worth logging the current state of the job, but returning it to the caller may not be a good idea. Maybe simply respond that it's not in the `initializing` state and do not disclose the current state?

File: src/Microsoft.Crank.Agent/Controllers/JobsController.cs
Comment (mkArtakMSFT) [inline src/Microsoft.Crank.Agent/Controllers/JobsController.cs:373]: What if some unexpected error happens above? I'd suggest wrapping lines 362-371 in a try - finally block and moving the `Delete` into a finally block.

**Review by mkArtakMSFT — CHANGES_REQUESTED** [review]
Just one small change to avoid a situation where the storage gets filled up.

File: src/Microsoft.Crank.Agent/Controllers/JobsController.cs
Comment (sebastienros) [reply src/Microsoft.Crank.Agent/Controllers/JobsController.cs:357]: It could be important for debugging to understand at what point the client tried to upload a file and how we got into this state. There is nothing to hide here.

File: src/Microsoft.Crank.Agent/Controllers/JobsController.cs
Comment (sebastienros) [reply src/Microsoft.Crank.Agent/Controllers/JobsController.cs]: Done

**Review by mkArtakMSFT — APPROVED** [review]

**Pull request merged**
